tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_get_default_backend_configuration.py

```diff
@@ -0,0 +1,263 @@
+        nwbfile = io.read()
+
+        raw_array = np.array([[11, 21, 31], [41, 51, 61]], dtype="int32")
+        raw_time_series = mock_TimeSeries(name="NewRawTimeSeries", data=raw_array)
+        nwbfile.add_acquisition(raw_time_series)
+
+        number_of_epochs = 5
+        for start_time, stop_time in zip(
+            np.linspace(start=0.0, stop=10.0, num=number_of_epochs),
+            np.linspace(start=1.0, stop=11.0, num=number_of_epochs),
+        ):
+            nwbfile.add_epoch(start_time=start_time, stop_time=stop_time)
+
+        ecephys_module = get_module(nwbfile=nwbfile, name="ecephys")
+        processed_array = np.array([[7.1, 8.1], [9.1, 10.1], [11.1, 12.1], [13.1, 14.1]])
+        processed_time_series = mock_TimeSeries(name="NewProcessedTimeSeries", data=processed_array)
+        ecephys_module.add(processed_time_series)
+
+        backend_configuration = get_default_backend_configuration(nwbfile=nwbfile, backend="zarr")
+
+    assert isinstance(backend_configuration, ZarrBackendConfiguration)
+
+    dataset_configurations = backend_configuration.dataset_configurations
+    assert len(dataset_configurations) == 4
+    assert "acquisition/NewRawTimeSeries/data" in dataset_configurations
+    assert "epochs/start_time/data" in dataset_configurations
+    assert "epochs/stop_time/data" in dataset_configurations
+    assert "processing/ecephys/NewProcessedTimeSeries/data" in dataset_configurations
+
+    # Best summary test of expected output is the printout
+    with patch("sys.stdout", new=StringIO()) as stdout:
+        print(backend_configuration)
+
+    expected_print = """
+Configurable datasets identified using the zarr backend
+-------------------------------------------------------
+
+epochs/start_time/data
+----------------------
+  dtype : float64
+  full shape of source array : (5,)
+  full size of source array : 0.00 GB
+
+  buffer shape : (5,)
+  maximum RAM usage per iteration : 0.00 GB
+
+  chunk shape : (5,)
+  disk space usage per chunk : 0.00 MB
+
+  compression method : gzip
+
+
+epochs/stop_time/data
+---------------------
+  dtype : float64
+  full shape of source array : (5,)
+  full size of source array : 0.00 GB
+
+  buffer shape : (5,)
+  maximum RAM usage per iteration : 0.00 GB
+
+  chunk shape : (5,)
+  disk space usage per chunk : 0.00 MB
+
+  compression method : gzip
+
+
+acquisition/NewRawTimeSeries/data
+---------------------------------
+  dtype : int32
+  full shape of source array : (2, 3)
+  full size of source array : 0.00 GB
+
+  buffer shape : (2, 3)
+  maximum RAM usage per iteration : 0.00 GB
+
+  chunk shape : (2, 3)
+  disk space usage per chunk : 0.00 MB
+
+  compression method : gzip
+
+
+processing/ecephys/NewProcessedTimeSeries/data
+----------------------------------------------
+  dtype : float64
+  full shape of source array : (4, 2)
+  full size of source array : 0.00 GB
+
+  buffer shape : (4, 2)
+  maximum RAM usage per iteration : 0.00 GB
```

Comment:
One more think!
I think that maximum ram usage per iteration is a very strong promise. 
At least I know that promise IS NOT hold if you are working in spikeinterface. And I would not be suprised if that fails for other interfaces.

I would changes it to the lesser commitment of "expected RAM memory ussage" and also give the units in GiB as that is the standard for memory.

Reply:
'expected' would be better, I agree

> At least I know that promise IS NOT hold if you are working in spikeinterface.

Can you give specific example here so I'm at least aware? Known cases where the `__init__` of an extractor reads entire array into memory? I was hoping every RecordingExtractor was fully lazy the whole way through

Reply:
It is not about loading the whole array, with that we are very careful I feel.

It is more than the wrong copy, casting or tranposition might trigger a memory allocation at some point of the extraction chain. This will generate memory allocations that within steps of the chain will produce ~2x3 times more ram consumption that they should based on chunk size alone and then there is this issue of **when** exactly that extra memory is liberated which is a topic that I don't understand fully an is OS dependent.

At some point I want to build general tests for the extractors to test that they don't have larger memory allocations but as usual, no time for that yet : (

See two relevant issues:
https://github.com/NeuralEnsemble/python-neo/pull/1265
https://github.com/SpikeInterface/spikeinterface/issues/2029

Reply:
Yes, OK I see...

That makes complete sense